herddb-core/src/main/java/herddb/file/FileCommitLog.java

```diff
@@ -141,13 +147,16 @@ private CommitFileWriter(long ledgerId, long sequenceNumber) throws IOException
             writtenBytes = 0;
         }
 
-        public void writeEntry(long seqnumber, LogEntry edit) throws IOException {
+        public int writeEntry(long seqnumber, byte[] entry) throws IOException {
 
             this.out.writeByte(ENTRY_START);
             this.out.writeLong(seqnumber);
-            int written = edit.serialize(this.out);
```

Comment:
entry now is serialized not in spool thread, but in the called thread.
I can't use ByteBuf because we have ExtendedDataOutputStream, which is not a "Channel" and it cannot deal with NIO Buffers (a mem copy is needed).